conans/test/remote/rest_api_test.py

```diff
@@ -49,6 +49,32 @@ def tearDownClass(cls):
     def tearDown(self):
         RestApiTest.server.clean()
 
+    def relative_url_completion_test(self):
```

Comment:
This might deserve an e2e test, using a test-server that actually returns relative urls.

Reply:
I would welcome an e2e test. The reason I didn't add them was exactly due to the necessary changes in the TestServer code. Any help there is greatly appreciated.

Reply:
No problem, we'll help with that test, we will contribute one e2e test